source/gitlab/gitlab.go

```diff
@@ -113,13 +113,30 @@ func WithInstance(client *gitlab.Client, config *Config) (source.Driver, error)
 }
 
 func (g *Gitlab) readDirectory() error {
-	nodes, response, err := g.client.Repositories.ListTree(g.projectID, g.listOptions)
-	if err != nil {
-		return err
-	}
+	var (
+		nodes    []*gitlab.TreeNode
+		n        []*gitlab.TreeNode
+		response *gitlab.Response
+		err      error
+	)
+
+	for {
+		n, response, err = g.client.Repositories.ListTree(g.projectID, g.listOptions)
```

Comment:
Use short declarations for readability. You're not going to see any perf differences since the return values are all pointers.